backend/headless/backend.c

```diff
@@ -84,6 +85,43 @@ static void handle_display_destroy(struct wl_listener *listener, void *data) {
 	backend_destroy(&backend->backend);
 }
 
+static bool check_rgba8_ext(void) {
+	const char *exts_str = (const char *)glGetString(GL_EXTENSIONS);
+	if (exts_str == NULL) {
+		wlr_log(WLR_ERROR, "glGetString(GL_EXTENSIONS) failed");
+		return false;
+	}
+	if (strstr(exts_str, "GL_OES_rgb8_rgba8") == NULL) {
+		wlr_log(WLR_ERROR, "GL_OES_rgb8_rgba8 not supported");
+		return false;
+	}
+	return true;
+}
+
+static bool backend_init(struct wlr_headless_backend *backend,
+		struct wl_display *display, struct wlr_renderer *renderer) {
+	wlr_backend_init(&backend->backend, &backend_impl);
+	backend->display = display;
+	wl_list_init(&backend->outputs);
+	wl_list_init(&backend->input_devices);
+
+	backend->renderer = renderer;
+	backend->egl = wlr_gles2_renderer_get_egl(renderer);
+
+	if (!wlr_egl_make_current(backend->egl, EGL_NO_SURFACE, NULL)) {
+		wlr_log(WLR_ERROR, "wlr_egl_make_current failed");
+		return false;
+	}
+	if (!check_rgba8_ext()) {
```

Comment:
Maybe fall back on a less efficient pixel format if the "good one" doesn't work? It's worse for `glReadPixels()` to not have the right format, but it's even worse if it doesn't work at all.